app/controllers/partners_controller.rb

```diff
@@ -164,6 +165,12 @@ def reactivate
 
   private
 
+  def validate_user
+    return unless current_user.has_role?(Role::PARTNER, current_partner)
+
+    redirect_to partner_user_root_path, error: "You must be logged in as the essentials bank's organization administrator to approve partner applications!"
```

Comment:
These 2 things don't seem to line up... the first line is checking if you are a partner user, but the text seems to indicate you need to be an org admin? I think this should be checking if you're an org admin if I understand the bug correctly.

Reply:
@dorner yes I tried putting check for org_admin but lots of test were failing where user was org_user, which I guess org_user also have access to this method, so I put check for both org_Admin and org_user
but I was wondering what if there is any other user too who has access to this

so instead I put check to block only partner, since so far what we know about this bug it's occurring when partner click on "Review this organization" link and try to access this method, we might need to update error message to include org_user also but I am not sure about the permission rules, wdyt?

Reply:
My understanding is that only the org admin can approve the requests. If that's the case, then if the current user is not signed in as an org\_admin, we should be kicking them out.

Reply:
You know,  I would have thought so, too -- but the "Approve Partner" button appears when I am signed in as user_1@example.com.

Reply:
@dorner,  @zeeshan-haidar What's more,  I was able to approve a partner at the user level.   The org_admin must be pretty much to limit user management -- this is relationship with existing partner management -- when we "approve" a partner,  what we are saying is "this partner is allowed to submit requests".

Reply:
@dorner, @zeeshan-haidar *However* I would still prefer to have it based on a 'we allow these roles to do this' basis,  rather than a 'we block these people from doing this'  -- I think it's better from a hygiene p.o.v -- If we, for instance,  latter put in a role for a kind of user with narrow access (for instance, a warehouse worker who would *only* be dealing with packing scheduled distributions), we're less likely to accidentally give them too many permissions.

Reply:
@cielf @dorner  as per office hour's discussion, only org_admin and org_user has been given permission to review partner information, also updated the text in email from "Review This Organization" to "Review this Partner"